### ISSUE TITLE
Document Archive

### ISSUE THREAD

**Issue by cubewhiz** [thewca/wca-documents#6]
There should be an archive of old documents for historical purposes.

We have already replaced/updated some motions, but I think the old documents should exist somewhere. In the documents themselves, though, there should be a note at the top (perhaps in bold red letters) saying something like - "Outdated Document - Replaced by xxx"

I imagine that some policy documents may eventually be replaced with new versions, too. In that case, the old version should be archived with such a note when the new version is implemented.

For now, I think at least the old Motions that were replaced should be created and archived somewhere.

**Comment by cubewhiz** [comment]
How does the WRC handle this? Would a similar implementation be possible?

**Comment by AlbertoPdRF** [comment]
The @thewca/wrc-team creates [releases](https://github.com/thewca/wca-regulations/releases), we could probably do the same!

**Comment by AlbertoPdRF** [comment]
I've done this already, see here: https://github.com/thewca/wca-documents/releases

For old policies, I think that keeping the PDFs on our Team Drives like we already do should be enough.

I'm closing this issue for now, but feel free to re-open it if you think there's still the need for it!

**Comment by cubewhiz** [comment]
I'm reopening this.

When creating new versions, old versions have broken links. I think we should keep the links to old versions as valid, but replace the document with a note at the top stating that the document is outdated.

**Comment by AlbertoPdRF** [comment]
@jonatanklosko any idea on how to handle this?

**Comment by jonatanklosko** [comment]
Hmm, I think it'd be hard to archive this way. Adding an `outdated` note would require the old documents to be rebuilt, not even sure at which point would that happen. Also if we do releases, then we would need to update the previous release whenever a new one emerges, and I have a feeling it's not easy to automate. So at this point I have no clever idea of dealing with this.

What would this actually solve? If someone downloads some previous version of documents, then it's clear they are outdated. Additionally if someone downloads the most recent release, and a new one happens, then he would still have the old documents locally without any kind of `outdated` note, or fixed links. Speaking of links, I don't think there's much we can do about them, they link to the documents on the WCA website, if we wanted an old document to reference other old documents, then we would probably need to store all the releases on the WCA website and make the links to point to a specific release, which sounds like an overkill to me (given it's just archiving).

To me there are two real cases:
1. Someone needs the documents at a specific point of time. -> He looks them up in releases.
2. Someone needs the latest documents. -> He either goes to the WCA website or downloads the latest release.

Correct me if these are not the only real cases =)

**Comment by cubewhiz** [comment]
In some cases, it will be because we have referenced a Motion with a link (e.g., an announcement) but the link is broken because the Motion was amended.

**Comment by danieljames-dj** [comment]
Can we have a permalink for every Motion? I'll explain with an example with the case of 'Candidate Delegates' (I know Candidate Delegate motion doesn't exist, but just for example):

Let the motion be 01.2010.1 - Candidate Delegates
The URL will be - https://www.worldcubeassociation.org/documents/motions/Candidate%20Delegates.pdf

Later, if we rename the motion to 'Junior Delegates', we can create a new URL accordingly and forward the old URL to the new URL.

Also, maybe we can give a link 'Archive' in our website which will go to https://github.com/thewca/wca-documents/releases so that they will be able to view all the old versions.

**Comment by cubewhiz** [comment]
I think that is a good idea.

**Comment by Jambrose777** [comment]
Agreed. This can also prevent issue like this that arise: https://github.com/thewca/wca-documents/issues/160

**Comment by cubewhiz** [comment]
A potential "problem" I can foresee, but someone with more knowledge correct me if I am wrong--the Motions appear on the page in alphabetical order, so renaming the files changes the order of all the documents on the page. Is that correct?

**Comment by danieljames-dj** [comment]
I was thinking of how to implement the solution. I would like to know is there any reason for having the motion number in filename? Since we already have it in the document, is it feasible to remove the motion number from the filename? If we do that, we can solve the problem of link getting broken.

**Comment by cubewhiz** [comment]
> I was thinking of how to implement the solution. I would like to know is there any reason for having the motion number in filename? Since we already have it in the document, is it feasible to remove the motion number from the filename? If we do that, we can solve the problem of link getting broken.

I like the idea. If we remove the motions numbers from the filenames, though, our Motions would be listed alphabetically instead of by Motion number. Does that seem like a problem?

**Comment by danieljames-dj** [comment]
If the order has to be maintained, maybe we can try to remove just numbers that represent year and month. Also, in the code we can try to remove the number that represents order as well.

Or another complicated solution will be to maintain a file where the order is mentioned and that order can be used to set the order.
